pkg/networkserver/grpc_deviceregistry.go

```diff
@@ -58,11 +58,9 @@ func (ns *NetworkServer) Set(ctx context.Context, req *ttnpb.SetEndDeviceRequest
 	var addDownlinkTask bool
 	dev, err := ns.devices.SetByID(ctx, req.EndDevice.EndDeviceIdentifiers.ApplicationIdentifiers, req.EndDevice.EndDeviceIdentifiers.DeviceID, gets, func(dev *ttnpb.EndDevice) (*ttnpb.EndDevice, []string, error) {
 		if dev != nil {
-			addDownlinkTask = ttnpb.HasAnyField(req.FieldMask.Paths,
-				"mac_state.device_class",
-				"queued_application_downlinks",
-			) && req.EndDevice.MACState.DeviceClass != ttnpb.CLASS_A &&
-				(len(req.EndDevice.QueuedApplicationDownlinks) > 0 || !req.EndDevice.MACState.CurrentParameters.Equal(req.EndDevice.MACState.DesiredParameters))
+			addDownlinkTask = ttnpb.HasAnyField(req.FieldMask.Paths, "mac_state.device_class") &&
```

Comment:
So we're creating a downlink task if the device class changes from A to something else AND there is existing downlink to send. Why do we no longer consider downlinks added to the queue by the current update (req)?

Reply:
Because we don't allow setting `application_downlink_queue` through RPC, see also https://github.com/TheThingsNetwork/lorawan-stack/blob/master/pkg/ttnpb/field_mask_validation.go#L490-L561